### PR TITLE
Add write metrics for kudo.

### PR DESCRIPTION
**Pull request by liurenjie1024**
This pr add support exporting write metrics of kudo serializer.

### CODE REVIEW

File: src/main/java/com/nvidia/spark/rapids/jni/kudo/SlicedBufferSerializer.java [hunk 1]
```diff
@@ -192,19 +194,28 @@ private long copySlicedData(HostColumnVectorCore column, SliceInfo sliceInfo) th
...
     }
   }
+
+  private long copyBufferAndPadForHost(HostMemoryBuffer buffer, long offset, long length) {
+    return withTime(() -> {
+      try {
+        writer.copyDataFrom(buffer, offset, length);
+        return padForHostAlignment(writer, length);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
```
Comment (jlowe) [inline src/main/java/com/nvidia/spark/rapids/jni/kudo/SlicedBufferSerializer.java:217]: Same remap concern here.

File: src/main/java/com/nvidia/spark/rapids/jni/kudo/SlicedBufferSerializer.java [hunk 2]
```diff
@@ -16,19 +16,19 @@
 
 package com.nvidia.spark.rapids.jni.kudo;
 
+import static com.nvidia.spark.rapids.jni.kudo.KudoSerializer.padForHostAlignment;
+import static com.nvidia.spark.rapids.jni.kudo.KudoSerializer.withTime;
```
Comment (jlowe) [inline src/main/java/com/nvidia/spark/rapids/jni/kudo/SlicedBufferSerializer.java:20]: Appears to be unused.
```suggestion
```

**Review by jlowe — APPROVED** [review]

File: src/main/java/com/nvidia/spark/rapids/jni/kudo/KudoSerializer.java [hunk 3]
```diff
@@ -176,7 +176,7 @@ public KudoSerializer(Schema schema) {
    * @param numRows   number of rows to write
    * @return number of bytes written
    */
-  long writeToStream(Table table, OutputStream out, int rowOffset, int numRows) {
+   WriteMetrics writeToStreamWithMetrics(Table table, OutputStream out, int rowOffset, int numRows) {
```
Comment (gerashegalov) [inline src/main/java/com/nvidia/spark/rapids/jni/kudo/KudoSerializer.java:179]: nit: should be 2-space indentation
```suggestion
  WriteMetrics writeToStreamWithMetrics(Table table, OutputStream out, int rowOffset, int numRows) {
```

File: src/main/java/com/nvidia/spark/rapids/jni/kudo/KudoSerializer.java [hunk 4]
```diff
@@ -195,6 +195,16 @@ long writeToStream(Table table, OutputStream out, int rowOffset, int numRows) {
...
   }
 
+  /**
+   * Write partition of an array of {@link HostColumnVector} to an output stream.
+   * See {@link #writeToStreamWithMetrics(HostColumnVector[], OutputStream, int, int)} for more
+   * details.
+   * @return number of bytes written
+   */
+  public long writeToStream(HostColumnVector[] columns, OutputStream out, int rowOffset, int numRows) {
+     return writeToStreamWithMetrics(columns, out, rowOffset, numRows).getWrittenBytes();
```
Comment (gerashegalov) [inline src/main/java/com/nvidia/spark/rapids/jni/kudo/KudoSerializer.java:205]: nit: inconsistent indentation with the rest of the file

**Pull request merged**